### ISSUE TITLE
Copy data to device (Documentation/Question)

### ISSUE THREAD

**Issue by jotsif** [google/jax#1989]
Hello !

I have a question about how Jax copies data the relevant device with regards to performance of XLA executions. My understanding here is vague so I might have misunderstood the issue but essentially:

When a piece of jax code is compiled and sent to the relevant device, how and when is the necessary tensors/arrays copied to that device? In pytorch for example you have the `.to(device) `calls which allows the user to control aspects of how this is done but i have not found a reference to an equivalent API for jax/xla. My current expectation/understanding is that jax copies the data to device for every xla-run when necessary.

My main issue is relevant for large datasets that are expensive to copy to a device and used for many different xla runs (for example reference datasets used in the loss function for each batch). If there is a possibility of precopy that data to the device and let the compiled code know where to find it would be highly beneficial for this kind of use case.

Thanks!

**Comment by jotsif** [comment]
`device_put` as in the quickstart documentation is probably what I want
